types/node/index.d.ts

```diff
@@ -6818,3 +6819,205 @@ declare module "http2" {
     export function connect(authority: string | url.URL, listener?: (session: ClientHttp2Session, socket: net.Socket | tls.TLSSocket) => void): ClientHttp2Session;
     export function connect(authority: string | url.URL, options?: ClientSessionOptions | SecureClientSessionOptions, listener?: (session: ClientHttp2Session, socket: net.Socket | tls.TLSSocket) => void): ClientHttp2Session;
 }
+
+declare module "perf_hooks" {
+	interface PerformanceEntry {
+		/**
+		 * The total number of milliseconds elapsed for this entry.
+		 * This value will not be meaningful for all Performance Entry types.
+		 */
+		readonly duration?: number;
+
+		/**
+		 * The name of the performance entry.
+		 */
+		readonly name?: string;
+
+		/**
+		 * The high resolution millisecond timestamp marking the starting time of the Performance Entry.
+		 */
+		readonly startTime?: number;
+
+		/**
+		 * The type of the performance entry.
+		 * Currently it may be one of: 'node', 'mark', 'measure', 'gc', or 'function'.
+		 */
+		readonly entryType?: string;
+
+		/**
+		 * When performanceEntry.entryType is equal to 'gc', the performance.kind property identifies
+		 * the type of garbage collection operation that occurred.
+		 * The value may be one of:
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL
+		 * - perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB
+		 */
+		readonly kind?: number;
+	}
+
+	interface PerformanceNodeTiming extends PerformanceEntry {
```

Comment:
I think all properties here are non optional. At least I haven't found any hint in the documentation in which case they are not present. A fast local test indicated that `0` is used to indicate no value.